### PR TITLE
Everything for the Department page (except for staff)

### LINKED ISSUE #179: Departments Landing Page
As a resident,
so that they can find the someone to provide a specific service,
they want to have a list of all the departments on a single page.

### PR DESCRIPTION
**Pull request by lynyedongze**
Resolves #179  and #180 

# Changes
Added a department tab that lists all the departments. Each department links to a dynamic page that admins could modify. A department has many services, and admins may edit, add, or delete service for a department in the edit page.

# Testing

1. Run `git checkout LynBranch`
2. Run `rails db:migrate`
3. Run `bundle install`
4. Run `rails test`, and make sure all test cases pass.
5. Start the server, and click on the department tab and create an arbitrary department.
6. Click on the department you just created and try to edit it. Make sure you (1) add some services, and (2) edit/ delete some service.

Important Note (credits to Dennis): 
- make sure you re-serve the server after cocoon install
- run `gem uninstall bundler --version 2.0.1` if you get a conflict message on when running `bundle ...`

### CODE REVIEW

**Comment by grinnelldennis** [comment]
Notes: 
- make sure you re-serve the server after cocoon install 
- run `gem uninstall bundler --version 2.0.1` if you get a conflict message on when running `bundle ...`

File: app/views/departments/_form.html.erb [hunk 1]
```diff
@@ -0,0 +1,51 @@
+<%= form_for @department do |f| %>
+<%= render 'shared/error_messages', object: f.object%>
+
+  <div class="form-horizontal", style="text-align:left">
+    <div class="form-group">
+
+      <div class="col-md-4 dept-col--contact">
```
[the remark this answers is not in the document]
Comment (lynyedongze) [reply app/views/departments/_form.html.erb]: We will assign a staff(object) to be the contact person of the department. The contact field in deparment page will show the related information about that staff.

File: app/controllers/services_controller.rb [hunk 2]
```diff
@@ -0,0 +1,74 @@
+class ServicesController < ApplicationController
```
[the remark this answers is not in the document]
Comment (lynyedongze) [reply app/controllers/services_controller.rb]: I think we should be able to add services on the create page now. I made the change a couple days ago. Would you want to check again?

File: app/views/departments/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,4 @@
+<h1 class="center">New Department</h1>
+  <div class="row">
+      <%= render 'form' %>
```
[the remark this answers is not in the document]
Comment (lynyedongze) [reply app/views/departments/new.html.erb]: All add service related features is included in the form partial that is rendered here

**Pull request merged**